### PR TITLE
Adds protos for multipart blob uploads

### PR DESCRIPTION
**Pull request by freider**
Implementation and usage in followup PR

### CODE REVIEW

**Comment by freider** [comment]
> Would it make more sense if it's the same method (`BlobCreate`) on the server, but the server returns a different response if multipart uploads are needed? We would have to send the file size to `BlobCreate` for this to make sense.

Perhaps, but a multipart upload requires *each part's* md5 checksum to be known ahead of time when creating the presigned urls so it would then require another roundtrip to the server with that data attached if we do it like that. Unless we feel checksumming of each individual part is unnecessary? We do still have checksumming for the full file, so perhaps that make sense?

**Comment by freider** [comment]
I think you might be right that there are advantages to having the server determine the upload specification that outweigh the need of using per-part MD5 checksumming. Refactored a bit

**Review by erikbern — APPROVED** [review]
looks good!

File: modal_proto/api.proto [hunk 1]
```diff
@@ -204,11 +204,22 @@ message BlobCreateRequest {
   // TODO(erikbern): how are these garbage collected?
   // Shouldn't they belong to an app?
   string content_md5 = 1;
+  string content_sha256_base64 = 2;
+  int64 content_length_bytes = 3;
```
Comment (erikbern) [inline modal_proto/api.proto:208]: micro nit, but i think of "bytes" as a default unit for anything like this, so it could be omitted

**Pull request merged**